model/monitorCache/implementation/DeliveryMonitoringData.php

```diff
@@ -200,12 +200,40 @@ private function getProgress()
         $session = $this->getTestSession();
 
         if ($session !== null) {
-            $pos = $session->getRoute()->getPosition();
-            $count = $session->getRouteCount();
-
             if ($session->isRunning()) {
-                $section = $session->getCurrentAssessmentSection();
-                $result =  __('%1$s - item %2$s/%3$s', $section->getTitle(), $pos+1, $count);
+                $route = $session->getRoute();
+                $routeItems = $route->getAllRouteItems();
+                $currentPosition = $route->getPosition();
+                $currentSection = $session->getCurrentAssessmentSection();
+                $currentSectionId = $currentSection->getIdentifier();
+
+                $sectionCount = 0;
+                $positionInSection = 0;
+                $offset = 0;
+                $offsetSection = 0;
+                $lastSection = null;
+
+                foreach ($routeItems as $routeItem) {
+                    $sections = $routeItem->getAssessmentSections();
+                    $sectionId = key(current($sections));
```

Comment:
Hi @jsconan,  
May be it will be safer to do something like `$sections->key()` here? 
As far as I now using `array` related functions leads to different behaviour, so you are getting first object property and passing that value to `key()` function. Hopefully here we have the only property as for now that contains collected values itself  so it works, but its never call `Iterators` methods behind. 
BTW: such construction works differently in PHP7